src/modules/hip/kernel/glitch.hpp

```diff
@@ -0,0 +1,319 @@
+                                      unsigned int *yOffsetR,
+                                      unsigned int *xOffsetG,
+                                      unsigned int *yOffsetG,
+                                      unsigned int *xOffsetB,
+                                      unsigned int *yOffsetB,
+                                      RpptROIPtr roiTensorPtrSrc)
+{
+
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    int xR, yR, xG, yG, xB, yB;
+    xR = id_x + xOffsetR[id_z];
+    yR = id_y + yOffsetR[id_z];
+    xG = id_x + xOffsetG[id_z];
+    yG = id_y + yOffsetG[id_z];
+    xB = id_x + xOffsetB[id_z];
+    yB = id_y + yOffsetB[id_z];
+
+    uint srcIdx, dstIdx;
+    srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+    dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    uint srcIdxR, srcIdxG, srcIdxB;
+    srcIdxR = srcIdx;
+    srcIdxG = srcIdxR + srcStridesNCH.y;
+    srcIdxB = srcIdxG + srcStridesNCH.y;
+
+    if ((yR >= 0) && (yR < roiTensorPtrSrc[id_z].xywhROI.roiHeight) && (xR >= 0) && (xR < roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        srcIdxR = (id_z * srcStridesNCH.x) + ((yR + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (xR + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+
+    if ((yG >= 0) && (yG < roiTensorPtrSrc[id_z].xywhROI.roiHeight) && (xG >= 0) && (xG < roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        srcIdxG = (id_z * srcStridesNCH.x) + ((yG + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (xG + roiTensorPtrSrc[id_z].xywhROI.xy.x) + srcStridesNCH.y;
+
+    if ((yB >= 0) && (yB < roiTensorPtrSrc[id_z].xywhROI.roiHeight) && (xB >= 0) && (xB < roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        srcIdxB = (id_z * srcStridesNCH.x) + ((yB + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (xB + roiTensorPtrSrc[id_z].xywhROI.xy.x) + 2 * srcStridesNCH.y;
+
+    dstPtr[dstIdx] = srcPtr[srcIdxR];
+    dstPtr[dstIdx + 1] = srcPtr[srcIdxG];
+    dstPtr[dstIdx + 2] = srcPtr[srcIdxB];
+}
+
+
+template <typename T>
+RppStatus hip_exec_glitch_tensor(T *srcPtr,
+                                     RpptDescPtr srcDescPtr,
+                                     T *dstPtr,
+                                     RpptDescPtr dstDescPtr,
+                                     RpptROIPtr roiTensorPtrSrc,
+                                     RpptRoiType roiType,
+                                     rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+    int localThreads_x = LOCAL_THREADS_X;
```

Comment:
These 3 local threads lines should be removed and LOCAL_THREADS_X should be used directly. Pls check other functions